### PR TITLE
Fix for #5262 - Short lived threads can cause exceptions

### PR DESCRIPTION
**Pull request by armadillu**
Try-catch around ofThread deteach(), and only detach if not joinable

### CODE REVIEW

**Comment by arturoc** [comment]
can you change it to just:

```
try{
	thread.detach();
}catch(...){}
```

if it's something that just happens and is really not an error we don't want an error to show up in the console and checking for joinable is not very safe in any case so better to just try to detach it and ignore the exception i think

**Comment by armadillu** [comment]
yep

**Comment by arturoc** [comment]
can you remove the check for `joinable()` too? it doesn't even make sense to join a thread from itself so an implementation might even return false there if called from the same thread

**Comment by arturoc** [comment]
thanks!
